### DISCUSSION TITLE
Jumpy tab width

### DISCUSSION THREAD

**Discussion by Mrsash2020** [onemen/TabMixPlus#328] [General]
I updated from 1.20 to 1.22 TMP and changed a few things but a long time annoyance has come back. Almost had forgotten it existed a while ago. Not sure if its firefox or TMP. The width of multirow tabs keeps changing. 

So lets say its stable. I close a tab which instigates an adjustment in the tab width of other tabs and it starts widening and shorting those other tabs until I close another tab.

Any thoughts?

**Comment by onemen** [comment]
Attach a file with your Tab Mix Plus preferences.

Can you attach a screenshot ?

Does it happens when you have more then one row, does `new tab button`  jump as well from after last tab to the side?

**Reply by Mrsash2020** [reply]
Basically when I go more than a row. A new tab, closing tabs all trigger this issue but yes going over a row. I also noticed just noticed I am missing the new tab button after the last tab. I dont remember exactly but was quite sure the plus which is now on the top row was after the last tab or was there a separate button. Arrgh cant remember. 

https://github.com/user-attachments/assets/f09d1f4c-6966-46b6-bea0-bfe36006b396


[TMPpref 2.txt](https://github.com/user-attachments/files/17029203/TMPpref.2.txt)

update: yeah so the new tab button in pref is set at after last tab but when you go over a row it does not move with it.

**Reply by onemen** [reply]
@Mrsash2020, 

try [Development Build](https://github.com/onemen/TabMixPlus/releases/tag/dev-build) maybe it fixed this issue.

i see message the the video file is corrupt.

Are you sure you are using version 1.22, the last version is 1.23

**Reply by Mrsash2020** [reply]
https://www.dropbox.com/scl/fi/ak20a8qsi55n1kiwsjk84/20240917_233524.mp4?rlkey=guivdn9j804uypwj790utio6b&st=tqj8b5h0&dl=0

I have a download in progress, so will update as soon as it finishes. lol long download, no resume option. Meanwhile check out the video in my dropbox. Will delete once you confirm you have downloaded. 

thanks

**Reply by onemen** [reply]
@Mrsash2020, 

I've downloaded the video, so far i did not manage to reproduce this issue.

Report later if it work better with the latest version

**Reply by Mrsash2020** [reply]
Will do. thanks. I did just remove the new tab button entirely and it does not jump now with the quick test. So it might be that going from one row to two it tried to move the new tab button to the last tab which is now on row 2 and as it fails, it bounces around? 

Will test more tomorrow morning. Going to bed now. thanks

**Reply by onemen** [reply]
try this [test build](https://github.com/user-attachments/files/17031973/test-build-1.23.1-20240917.1923.zip) with your original preferences.

download the file, rename it to .xpi and install it.

it suppose to prevent the jump when `new tab button` is after last tab, it should keep the button on the right side when the new row have one tab.

**Reply by Mrsash2020** [reply]
It seems 1.23 version has now fixed it for me. After I installed it, I still had to remove and then add the new tab button at the end. Doing some testing now. Will post back.

Hmm. Hold that thought. The new tab button moved again and it starts all over again. May try this test build.

**Comment by Brian-Westerman** [comment]
I have this same problem when I change the size of my FireFox screen.  I think you will find that if you adjust your screen width by clicking the left or right side and pulling it in or out just slightly smaller or larger, that it will go away and you won’t see it again.  It appears to be just something that happens to me when I have changed my firefox screen and I just happen to get a size that causes that same thing to happen.

**Reply by Mrsash2020** [reply]
Yes I have noticed that as well previously but it with this adjustment it can again become a problem. I resize the firefox window size and it would become stable until that is I start either adding or deleting tabs. Then another adjustment is required which is annoying when you are doing a lot of browsing.

I think the issue is purely the add new tab option after the last tab. I like this option so happy now it seems its working after 1.23 update and then remove add new tab button after last tab > apply. Then add the option > apply.

**Comment by onemen** [comment]
I am working on a fix. Until it will be ready just set `new tab button` to be on right side instead of after tabs

**Reply by Mrsash2020** [reply]
ok cool. Did you want me to try the one in the previous post? Sorry had guests pop in.

Update: I could be wrong but it seems to be fixed with firefox update I did yesterday. Prob will do more testing. Still on 1.23 TMP
Nope. Spoke too soon

**Comment by onemen** [comment]
Can you test this new build, I have update the fix from last year

Download this [test build](https://github.com/user-attachments/files/17189492/test-build-1.23.1-20240930.1506.zip)

Rename it to `.xpi` and install it.

Report back.

**Reply by Mrsash2020** [reply]
Ok before I started it had stopped being jumpy since the Firefox update I mentioned, however what had not changed is the New Tab Button(NTB) behavior. It kept going back to top right at the edge of the first row. 
I installed the test build and tried a few things

1 Adding tabs row 1 then 2 then three. The NTB kept moving correctly to the end of third line.
2 Closing and adding tabs. Closing tabs from row three to row two caused the NTB button to go to row one and not at the end of row two.
3 Adjusting the width of the Firefox window did the same

2 & 3 are intermittent issues......
![Screenshot 2024-09-30 224734](https://github.com/user-attachments/assets/25684b2f-69cb-42d7-883c-75c2f33b0719)

![Screenshot 2024-09-30 224917](https://github.com/user-attachments/assets/ee9a5be8-cf57-47c2-9362-84c170957467)

Also lost a few options from the right click menu. I got them back once I disabled and enabled TMP. Not sure if any lasting issues as a result.

![image](https://github.com/user-attachments/assets/c66cfaae-7cbc-4f7c-9565-c720afcda721)

![image](https://github.com/user-attachments/assets/79fd9019-59e0-4750-960a-4873f8562c21)

**Reply by Mrsash2020** [reply]
Also seem to have lost my close tab button.

**Reply by onemen** [reply]
- Open about:support page and click "Clear startup cache…" to force Firefox to load the the installed Tab Mix version
- what do you mean by `seem to have lost my close tab button`, any messages in the error console?, post your latest Tab Mix preferences, and screenshot of this issue

**Reply by Mrsash2020** [reply]
Ok just did the clear cache and it brought back the close tab button on the right so all good there. Quick test of tab jumping and new tab button moving around seems its fixed but need to test more I think. Will get back ASAP.
thanks

**Reply by Mrsash2020** [reply]
Sorry took some time. The jumping bit is sorted I think. The new tab button aligns with the number of rows you specify. So number of rows = 6. If you open more tabs it creates 7th row just fine but the new tab button switches to being on the right top next to "X".
Not a major issue really just reporting back. If you close enough tabs to say drop to 6 rows the new tab button then moves back to the after the last tab position as expected.

**Reply by onemen** [reply]
I placed the `new tab button` on the top right to avoid the `new tab button` to be on the 7th row alone, when the button is after tabs there is place to one more tab is the 6th row.
